kv/txn_coord_sender.go

```diff
@@ -769,6 +769,8 @@ func (tc *TxnCoordSender) updateState(ctx context.Context, ba roachpb.BatchReque
 				// continue its transaction after receiving ConditionFailedError, which
 				// can come from a unique index violation.
 			}
+			newTxn.Update(pErr.GetTxn())
+			pErr.SetTxn(newTxn)
```

Comment:
it looks like we're doing (or at least should be) in every branch. Can we just move it outside, after the type switch?